test/deprecated/owned-shared-from-borrow.chpl

```diff
@@ -0,0 +1,28 @@
+// This test shows a pattern that is allowed in 1.18
```

Comment:
Are you planning on adding a "deprecated" warning for these cases?

Reply:
I'm concerned about doing so at this late date & have a full plate with spec updates. If you wanted to try to do so though that'd be fine with me. I was planning to just remove them without a warning, on the theory that Owned/Shared weren't used much in their previous form.